src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -4367,10 +4367,10 @@ void SmartScript::ProcessEvent(SmartScriptHolder& e, Unit* unit, uint32 var0, ui
                 {
                     if (IsPlayer(target))
                         playerCount++;
-
-                    if (playerCount >= e.event.nearPlayer.minCount)
-                        ProcessAction(e, target->ToUnit());
                 }
+                    if (playerCount >= e.event.nearPlayer.minCount)
+                        ProcessAction(e, unit);
+
```

Comment:
Several things here.

1. What was broken in the first place?
2. By doing this you are making the for loop above do nothing
3. The ident is wrong